### DISCUSSION TITLE
Nullish coalescing operator in the cdn min.js build

### DISCUSSION THREAD

**Discussion by moshrepo00** [alpinejs/alpine#3619] [7. Bugs (archived)]
I've noticed the presence of the Nullish coalescing operator (??) in one location in the latest alpinejs distribution: https://cdn.jsdelivr.net/npm/alpinejs@3.12.2/dist/cdn.min.js . Merely removing it in that one location would get alpinejs to work on a number of older browsers.  I'm not sure why it wasn't removed in the final bundle. 

Also, thanks for this great js library, absolutely love using it!

**Comment by ekwoka** [comment]
I'm surprised jsdelivr doesn't have bundler control options. `esm.sh` does if you want to use that one. https://esm.sh/v125/alpinejs@3.12.2/es2019/dist/cdn.js for example. Hell, ESM.sh will auto target what the browser can do, so you don't even need to specify. Could just do https://esm.sh/v125/alpinejs@3.12.2/dist/cdn.js and it'll transpile if needed.

Generally, targeting things is in the realm of consumer bundling, and `??` is transpilable so you can bundle your scripts to the target you want. Better than holding the entire web back.

Instead of being concerned with squashing all these, it makes more sense to adjust the bundle scripts for the CDN options to target es2017 (I believe this is the absolute minimum required in terms of what can't be transpiled) but 2019 would transpile `r.detail??r.target.value` to `if(r instanceof CustomEvent&&r.detail!==void 0)return(i=r.detail)!=null?i:r.target.value`.

Nullish coalescing is supported in 93.6% of users. `AsyncFunction` is at 95%. So the difference for this change has a ceiling of 1.4%...

You can make a PR (or maybe I'll get to it) that sets a target on the CDN builds, and we'll see what makes the most sense to Caleb.

**Comment by ekwoka** [comment]
#3625
